ComponentKit/Core/CKComponentController.mm

```diff
@@ -66,17 +68,37 @@ static void eraseAnimation(CKAppliedComponentAnimationMap &map, CKComponentAnima
   }
 }
 
+struct CKComponentControllerAnimationData {
+  CKComponentAnimationID nextAnimationID;
+  std::vector<CKPendingComponentAnimation> pendingAnimationsOnInitialMount;
+  CKAppliedComponentAnimationMap appliedAnimationsOnInitialMount;
+  std::vector<CKPendingComponentAnimation> pendingAnimations;
+  CKAppliedComponentAnimationMap appliedAnimations;
+};
+
+struct CKComponentControllerAnimationWrapper {
+public:
+  CKComponentControllerAnimationData *operator->() {
+    if (_animationData == nullptr) {
+      _animationData.reset(new CKComponentControllerAnimationData());
+    }
+    return _animationData.get();
+  };
```

Comment:
Seems like a nice way to make this lazy with minimal complexity in the `CKComponentController` implementation. I like this approach.